pkg/operation/botanist/component/seedadmissioncontroller/seedadmissioncontroller.go

```diff
@@ -405,6 +421,21 @@ func init() {
 	utilruntime.Must(err)
 }
 
+func (g *gardenerSeedAdmissionController) getReplicas(ctx context.Context) (int32, error) {
+	nodeList := &corev1.NodeList{}
+	err := g.client.List(ctx, nodeList)
+	if err != nil {
+		return 0, err
+	}
```

Comment:
As you are only interested in the number of items in the list, can you use a `PartialObjectMetadataList` like here:

https://github.com/gardener/gardener/blob/5988d5ea7bda62f66253ad473f154be814a8e5b3/pkg/admissioncontroller/webhooks/admission/internaldomainsecret/admission.go#L154-L169